File: app/src/at-phutran/java/asiantech/internship/summer/retrofit/RealPathUtil.java
package asiantech.internship.summer.retrofit;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.CursorLoader;
import android.database.Cursor;
import android.net.Uri;
import android.provider.MediaStore;

class RealPathUtil {
    @SuppressLint("NewApi")
    static String getRealPathFromURI_API11to18(Context context, Uri contentUri) {
        String[] array = {MediaStore.Images.Media.DATA};
        String result = null;
        CursorLoader cursorLoader = new CursorLoader(context, contentUri, array, null, null, null);
        Cursor cursor = cursorLoader.loadInBackground();
        if (cursor != null) {
            int column_index =
                    cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA);
            cursor.moveToFirst();
            result = cursor.getString(column_index);
        }
        return result;
    }
}

File: app/src/at-phutran/java/asiantech/internship/summer/kotlin/retrofit/RealPathUtil.kt
package asiantech.internship.summer.kotlin.retrofit

import android.annotation.SuppressLint
import android.content.Context
import android.content.CursorLoader
import android.net.Uri
import android.provider.MediaStore

class RealPathUtil {
    @SuppressLint("NewApi")
    fun getRealPathFromUriAPI11to18(context: Context, contentUri: Uri): String? {
        val array = arrayOf(MediaStore.Images.Media.DATA)
        var result: String? = null
        val cursorLoader = CursorLoader(context, contentUri, array, null, null, null)
        val cursor = cursorLoader.loadInBackground()
        if (cursor != null) {
            val indexColumn = cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA)
            cursor.moveToFirst()
            result = cursor.getString(indexColumn)
        }
        return result
    }
}